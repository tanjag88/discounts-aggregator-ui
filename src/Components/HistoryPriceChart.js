import React from "react";
import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend,
} from "recharts";

function HistoryPriceChart({ priceHistory }) {
  const priceHistoryDataPoints = priceHistory.map((p) => ({
    date: new Date(p.date).toLocaleDateString("en-US", {
      day: "numeric",
      month: "short",
      year: "numeric",
    }),
    price: p.amount,
  }));

  return (
    <>
      <LineChart
        width={500}
        height={300}
        data={priceHistoryDataPoints}
        margin={{
          top: 5,
          right: 30,
          left: 20,
          bottom: 5,
        }}
      >
        <CartesianGrid strokeDasharray="5 5" />
        <XAxis dataKey="date" />
        <YAxis />

        <Tooltip />
        <Legend />
        <Line
          name="Price in CA$"
          type="monotone"
          dataKey="price"
          stroke="#8884d8"
          fill="#8884d8"
          activeDot={{ r: 8 }}
        />
      </LineChart>
    </>
  );
}
export default HistoryPriceChart;
