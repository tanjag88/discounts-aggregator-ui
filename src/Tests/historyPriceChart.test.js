import React from "react";
import renderer from "react-test-renderer";
import HistoryPriceChart from "../Components/HistoryPriceChart";

test("renders correctly", () => {
  const priceHistory = [
    {
      amount: 180,
      date: 1606861107000,
    },
    {
      amount: 140,
      date: 1612217907000,
    },
    {
      amount: 130,
      date: 1614637107000,
    },
  ];

  const tree = renderer
    .create(<HistoryPriceChart priceHistory={priceHistory} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
