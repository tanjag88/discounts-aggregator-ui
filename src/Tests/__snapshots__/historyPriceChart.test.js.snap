// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="recharts-wrapper"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "cursor": "default",
      "height": 300,
      "position": "relative",
      "width": 500,
    }
  }
>
  <svg
    className="recharts-surface"
    height={300}
    version="1.1"
    viewBox="0 0 500 300"
    width={500}
  >
    <defs>
      <clipPath
        id="recharts1-clip"
      >
        <rect
          height={260}
          width={390}
          x={80}
          y={5}
        />
      </clipPath>
    </defs>
    <g
      className="recharts-cartesian-grid"
    >
      <g
        className="recharts-cartesian-grid-horizontal"
      >
        <line
          fill="none"
          height={260}
          offset={
            Object {
              "bottom": 35,
              "brushBottom": 35,
              "height": 260,
              "left": 80,
              "right": 30,
              "top": 5,
              "width": 390,
            }
          }
          stroke="#ccc"
          strokeDasharray="5 5"
          width={390}
          x={80}
          x1={80}
          x2={470}
          y={5}
          y1={265}
          y2={265}
        />
        <line
          fill="none"
          height={260}
          offset={
            Object {
              "bottom": 35,
              "brushBottom": 35,
              "height": 260,
              "left": 80,
              "right": 30,
              "top": 5,
              "width": 390,
            }
          }
          stroke="#ccc"
          strokeDasharray="5 5"
          width={390}
          x={80}
          x1={80}
          x2={470}
          y={5}
          y1={200}
          y2={200}
        />
        <line
          fill="none"
          height={260}
          offset={
            Object {
              "bottom": 35,
              "brushBottom": 35,
              "height": 260,
              "left": 80,
              "right": 30,
              "top": 5,
              "width": 390,
            }
          }
          stroke="#ccc"
          strokeDasharray="5 5"
          width={390}
          x={80}
          x1={80}
          x2={470}
          y={5}
          y1={135}
          y2={135}
        />
        <line
          fill="none"
          height={260}
          offset={
            Object {
              "bottom": 35,
              "brushBottom": 35,
              "height": 260,
              "left": 80,
              "right": 30,
              "top": 5,
              "width": 390,
            }
          }
          stroke="#ccc"
          strokeDasharray="5 5"
          width={390}
          x={80}
          x1={80}
          x2={470}
          y={5}
          y1={70}
          y2={70}
        />
        <line
          fill="none"
          height={260}
          offset={
            Object {
              "bottom": 35,
              "brushBottom": 35,
              "height": 260,
              "left": 80,
              "right": 30,
              "top": 5,
              "width": 390,
            }
          }
          stroke="#ccc"
          strokeDasharray="5 5"
          width={390}
          x={80}
          x1={80}
          x2={470}
          y={5}
          y1={5}
          y2={5}
        />
      </g>
      <g
        className="recharts-cartesian-grid-vertical"
      >
        <line
          fill="none"
          height={260}
          offset={
            Object {
              "bottom": 35,
              "brushBottom": 35,
              "height": 260,
              "left": 80,
              "right": 30,
              "top": 5,
              "width": 390,
            }
          }
          stroke="#ccc"
          strokeDasharray="5 5"
          width={390}
          x={80}
          x1={80}
          x2={80}
          y={5}
          y1={5}
          y2={265}
        />
        <line
          fill="none"
          height={260}
          offset={
            Object {
              "bottom": 35,
              "brushBottom": 35,
              "height": 260,
              "left": 80,
              "right": 30,
              "top": 5,
              "width": 390,
            }
          }
          stroke="#ccc"
          strokeDasharray="5 5"
          width={390}
          x={80}
          x1={275}
          x2={275}
          y={5}
          y1={5}
          y2={265}
        />
        <line
          fill="none"
          height={260}
          offset={
            Object {
              "bottom": 35,
              "brushBottom": 35,
              "height": 260,
              "left": 80,
              "right": 30,
              "top": 5,
              "width": 390,
            }
          }
          stroke="#ccc"
          strokeDasharray="5 5"
          width={390}
          x={80}
          x1={470}
          x2={470}
          y={5}
          y1={5}
          y2={265}
        />
      </g>
    </g>
    <g
      className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
    >
      <line
        className="recharts-cartesian-axis-line"
        fill="none"
        height={30}
        orientation="bottom"
        stroke="#666"
        type="category"
        width={390}
        x={80}
        x1={80}
        x2={470}
        y={265}
        y1={265}
        y2={265}
      />
      <g
        className="recharts-cartesian-axis-ticks"
      >
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={30}
            orientation="bottom"
            stroke="#666"
            type="category"
            width={390}
            x={80}
            x1={80}
            x2={80}
            y={265}
            y1={271}
            y2={265}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={30}
            orientation="bottom"
            stroke="none"
            textAnchor="middle"
            type="category"
            width={390}
            x={80}
            y={273}
          >
            <tspan
              dy="0.71em"
              x={80}
            >
              Dec 1, 2020
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={30}
            orientation="bottom"
            stroke="#666"
            type="category"
            width={390}
            x={80}
            x1={275}
            x2={275}
            y={265}
            y1={271}
            y2={265}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={30}
            orientation="bottom"
            stroke="none"
            textAnchor="middle"
            type="category"
            width={390}
            x={275}
            y={273}
          >
            <tspan
              dy="0.71em"
              x={275}
            >
              Feb 1, 2021
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={30}
            orientation="bottom"
            stroke="#666"
            type="category"
            width={390}
            x={80}
            x1={470}
            x2={470}
            y={265}
            y1={271}
            y2={265}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={30}
            orientation="bottom"
            stroke="none"
            textAnchor="middle"
            type="category"
            width={390}
            x={470}
            y={273}
          >
            <tspan
              dy="0.71em"
              x={470}
            >
              Mar 1, 2021
            </tspan>
          </text>
        </g>
      </g>
    </g>
    <g
      className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
    >
      <line
        className="recharts-cartesian-axis-line"
        fill="none"
        height={260}
        orientation="left"
        stroke="#666"
        type="number"
        width={60}
        x={20}
        x1={80}
        x2={80}
        y={5}
        y1={5}
        y2={265}
      />
      <g
        className="recharts-cartesian-axis-ticks"
      >
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={260}
            orientation="left"
            stroke="#666"
            type="number"
            width={60}
            x={20}
            x1={74}
            x2={80}
            y={5}
            y1={265}
            y2={265}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={260}
            orientation="left"
            stroke="none"
            textAnchor="end"
            type="number"
            width={60}
            x={72}
            y={265}
          >
            <tspan
              dy="0.355em"
              x={72}
            >
              0
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={260}
            orientation="left"
            stroke="#666"
            type="number"
            width={60}
            x={20}
            x1={74}
            x2={80}
            y={5}
            y1={200}
            y2={200}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={260}
            orientation="left"
            stroke="none"
            textAnchor="end"
            type="number"
            width={60}
            x={72}
            y={200}
          >
            <tspan
              dy="0.355em"
              x={72}
            >
              45
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={260}
            orientation="left"
            stroke="#666"
            type="number"
            width={60}
            x={20}
            x1={74}
            x2={80}
            y={5}
            y1={135}
            y2={135}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={260}
            orientation="left"
            stroke="none"
            textAnchor="end"
            type="number"
            width={60}
            x={72}
            y={135}
          >
            <tspan
              dy="0.355em"
              x={72}
            >
              90
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={260}
            orientation="left"
            stroke="#666"
            type="number"
            width={60}
            x={20}
            x1={74}
            x2={80}
            y={5}
            y1={70}
            y2={70}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={260}
            orientation="left"
            stroke="none"
            textAnchor="end"
            type="number"
            width={60}
            x={72}
            y={70}
          >
            <tspan
              dy="0.355em"
              x={72}
            >
              135
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={260}
            orientation="left"
            stroke="#666"
            type="number"
            width={60}
            x={20}
            x1={74}
            x2={80}
            y={5}
            y1={5}
            y2={5}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={260}
            orientation="left"
            stroke="none"
            textAnchor="end"
            type="number"
            width={60}
            x={72}
            y={5}
          >
            <tspan
              dy="0.355em"
              x={72}
            >
              180
            </tspan>
          </text>
        </g>
      </g>
    </g>
    <g
      className="recharts-layer recharts-line"
    >
      <path
        className="recharts-curve recharts-line-curve"
        clipPath={null}
        d="M80,5C145,29.07407407407407,210,53.14814814814813,275,62.77777777777777C340,72.4074074074074,405,74.81481481481482,470,77.22222222222223"
        fill="none"
        height={260}
        name="Price in CA$"
        points={
          Array [
            Object {
              "payload": Object {
                "date": "Dec 1, 2020",
                "price": 180,
              },
              "value": 180,
              "x": 80,
              "y": 5,
            },
            Object {
              "payload": Object {
                "date": "Feb 1, 2021",
                "price": 140,
              },
              "value": 140,
              "x": 275,
              "y": 62.77777777777777,
            },
            Object {
              "payload": Object {
                "date": "Mar 1, 2021",
                "price": 130,
              },
              "value": 130,
              "x": 470,
              "y": 77.22222222222223,
            },
          ]
        }
        stroke="#8884d8"
        strokeDasharray="0px 0px"
        strokeWidth={1}
        type="monotone"
        width={390}
      />
    </g>
  </svg>
  <div
    className="recharts-legend-wrapper"
    style={
      Object {
        "bottom": 5,
        "height": "auto",
        "left": 20,
        "position": "absolute",
        "width": 450,
      }
    }
  >
    <ul
      className="recharts-default-legend"
      style={
        Object {
          "margin": 0,
          "padding": 0,
          "textAlign": "center",
        }
      }
    >
      <li
        className="recharts-legend-item legend-item-0"
        style={
          Object {
            "display": "inline-block",
            "marginRight": 10,
          }
        }
      >
        <svg
          className="recharts-surface"
          height={14}
          style={
            Object {
              "display": "inline-block",
              "marginRight": 4,
              "verticalAlign": "middle",
            }
          }
          version="1.1"
          viewBox="0 0 32 32"
          width={14}
        >
          <path
            className="recharts-legend-icon"
            d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
            fill="none"
            stroke="#8884d8"
            strokeWidth={4}
          />
        </svg>
        <span
          className="recharts-legend-item-text"
          style={
            Object {
              "color": "#8884d8",
            }
          }
        >
          Price in CA$
        </span>
      </li>
    </ul>
  </div>
  <div
    className="recharts-tooltip-wrapper"
    style={
      Object {
        "MozTransform": "translate(undefinedpx, undefinedpx)",
        "OTransform": "translate(undefinedpx, undefinedpx)",
        "WebkitTransform": "translate(undefinedpx, undefinedpx)",
        "left": 0,
        "msTransform": "translate(undefinedpx, undefinedpx)",
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "transform": "translate(undefinedpx, undefinedpx)",
        "visibility": "hidden",
      }
    }
  >
    <div
      className="recharts-default-tooltip"
      style={
        Object {
          "backgroundColor": "#fff",
          "border": "1px solid #ccc",
          "margin": 0,
          "padding": 10,
          "whiteSpace": "nowrap",
        }
      }
    >
      <p
        className="recharts-tooltip-label"
        style={
          Object {
            "margin": 0,
          }
        }
      >
        
      </p>
    </div>
  </div>
</div>
`;
